### PR TITLE
[LPS-145244 & LPS-145245 & LPS-145246] [CP 2.0] Create Subscription Management Component

### PR DESCRIPTION
**Pull request by claricebrasil**

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.subscription-navbar {
+    border: 1px solid #c3c5c9;
+    height: 36px;
+
+    button {
+        padding: 0;
+    }
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss]:
```suggestion
```

Should be good use the `p-0` style class instead.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.subscription-navbar {
+    border: 1px solid #c3c5c9;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss]: This color should be present in style-book? or already exists.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    .hide {
+        display: none;
+    }
+}
+
+.subscription-navbar-dropdown {
+    .hide {
+        display: none;
+    }
+}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss:8]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+.btn-subscription-group-selected {
+    color: #004ad7 !important;
+    background-color: #fff;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss]: I guess that we can use the `bg-white` style class

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    &:focus {
+    background-color: #fff;
+    box-shadow: none;
+    border: 1px solid #004ad7;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss]: This color should be present in style-book? or already exists.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        display: none;
+    }
+}
+
+.btn-subscription-group-selected {
+    color: #004ad7 !important;
+    background-color: #fff;
+
+    &:focus {
+    background-color: #fff;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_subscription-navbar.scss]: This color should be present in style-book? or already exists.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/QuickLinksPanel/index.js [hunk 7]
```diff
@@ -39,8 +39,12 @@ const QuickLinksPanel = ({accountKey}) => {
 		);
 
 		if (quickLinksExpandedStorage) {
-			setExpandedPanel(JSON.parse(quickLinksExpandedStorage));
+			dispatch({
+				payload: JSON.parse(quickLinksExpandedStorage),
+				type: actionTypes.UPDATE_QUICK_LINKS_EXPANDED_PANEL,
+			});
 		}
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/QuickLinksPanel/index.js]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js [hunk 8]
```diff
@@ -9,35 +10,156 @@
...
+                    </ClayButton>
+                  }
+                >
+                  {subscriptionGroups.map((tag) => (
+                    <DropDown.Item
+                      key={tag.name}
+                      onClick={(event) => setSelectedSubscriptionGroup(event.target.value)}
+                      value={tag.name}
+                    >
+                      {tag.name}
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js]: No, subscriptionGroups only have an accountKey or a name

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js [hunk 9]
```diff
@@ -9,35 +10,156 @@
...
+
+    updateShowDropDown();
+    window.addEventListener("resize", updateShowDropDown);
+  }, [isQuickLinksExpanded]);
+
+ 
+
+  return (
+    <div className="d-flex rounded-pill w-100" ref={subscriptionNavbarRef}>
+      <nav className="mb-2 mt-4 pt-2">
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js]: Margin top should be bigger than margin bottom. That's not possible with my

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js [hunk 10]
```diff
@@ -9,35 +10,156 @@
...
+        {subscriptionGroups.length === 1 && subscriptionGroups.map((tag) => (
+          <span
+            className="h5 text-brand-primary"
+            key={tag.name}
+          >
+            {tag.name}
+          </span>
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js]: No, subscriptionGroups only have an accountKey or a name

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js [hunk 11]
```diff
@@ -9,35 +10,156 @@
...
+                {subscriptionGroups.map((tag) => (
+                  <button
+                    className={classNames('btn px-4 text-neutral-4 rounded-pill', {
+                      'btn-subscription-group': selectedButton !== tag.name, 
+                      'btn-subscription-group-selected label-primary': selectedButton === tag.name})}
+                    key={tag.name}
+                    onClick={handleClick}
+                    value={tag.name}
+                  >
+                    {tag.name}
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsNavbar/index.js]: No, subscriptionGroups only have an accountKey or a name.

**Comment by jairmedeiros** [comment]
Just starting reviewing :)

**Comment by jairmedeiros** [comment]
ci:test:sf

**Comment by jairmedeiros** [comment]
resent https://github.com/pt-liferay-solutions/liferay-portal/pull/736

**Pull request closed without merging**